### PR TITLE
Use Environment Variables

### PR DESCRIPTION
**Pull request by pastaq**
- Uses environment variables instead of hard coding the "deck" user/group. This
adds support for systems other than the steam deck that are using the DeckUI.

### CODE REVIEW

File: backend/browser.py
Comment (TrainDoctor) [inline backend/browser.py:53]: As already discussed this call to time.sleep(1) seems to break based upon how imported the time module. More testing can be done once this is resolved.

**Review by TrainDoctor — CHANGES_REQUESTED** [review]
Overall good changes that help to support an array of hardware rather than just a hardcoded deck user but needs the mentioned bug fix along with testing on both real Steam Deck hardware and on desktop linux as needed. Thanks for your PR and hopefully we can help you to get this running.

File: backend/browser.py
Comment (pastaq) [reply backend/browser.py]: Changed this and identified a few other import issues as well, specifically i was using the subprocess prefix and I missed the PIP import. Also used the asyncio sleep in main.py instead of time wince that is an async function which I missed the first push. Everything should be testable now.

**Review by TrainDoctor — APPROVED** [review]
Looks good to go. Turns out the uninstall prompt issues are in our code base so sorry @botatooo I'll be reviewing your code in that context to asap. I'll need another approving review from others before this can fully go ahead.

**Review by AAGaming00 — CHANGES_REQUESTED** [review]
The code for determining the current user should be made into a utility instead of duplicated across files.

File: backend/browser.py
Comment (TrainDoctor) [inline backend/browser.py:47]: Duplication of code. Please move to a utility method that can be invoked in browser and main.

File: backend/main.py
Comment (TrainDoctor) [inline backend/main.py:38]: As previously stated duplicated code.

File: backend/main.py
Comment (TrainDoctor) [inline backend/main.py:37]: Relating to previous issues found with .Popen, this terminal invocation would preferably be a set of python code. If not possible to use python then measures to ensure the prevention of a zombie process from .Popen should be taken.

**Review by TrainDoctor — CHANGES_REQUESTED** [review]
Upon further review and previous issues encountered with earlier versions relating to .Popen causing zombie processes we'd like to request that the terminal invocations be moved over to python code where possible. We are flexible if this isn't viable but more code to ensure the prevention of zombie processes will be required. And of course, as @AAGaming00  already stated it would be helpful if the code for finding the user be moved into a utility function in helpers.py.

Also, if possible caching the result of this would be desirable so that the function doesn't need to be ran after first boot. (This is desired but not required.)

File: backend/browser.py
Comment (pastaq) [reply backend/browser.py]: Done

File: backend/main.py
Comment (pastaq) [reply backend/main.py]: Done

File: backend/main.py
Comment (pastaq) [reply backend/main.py]: All code now uses call or check_output.

**Comment by pastaq** [comment]
Made requested changes. Also noticed some inefficiencies/duplication in imports so I ordered them alphanumerically and simplified when they were duplicated, moved all logic to after imports.

File: backend/main.py
Comment (JakobDev) [inline backend/main.py:24]:
```suggestion
HOME_PATH = os.path.expanduser("~")
```
It is possible to have a home directory outside /home

**Review by TrainDoctor — APPROVED** [review]
This appears good to merge. I'm waiting for testers to provide additional feedback and once they have provided me with feedback, I can proceed with merging if no problems are found during testing.
Thank you again for your time working on this.

File: backend/main.py
Comment (TrainDoctor) [reply backend/main.py:24]: While this is technically correct, we are focused on supporting primarily Steam Deck hardware + SteamOS 3. We will take steps to help support other distros but if their home directories do not exist withing ``/home`` then that's going to be exotic enough that we're not going to spend time supporting that. Apologies.

**Comment by TrainDoctor** [comment]
Currently this doesn't account for multiple users but since the deck is a single user (at a time) system this is a non-issue. If issues do arise on non-deck systems relating to multiple users being logged in, PRs will be accepted to resolve the issue.
Moving to merge this into main. Thanks for your contribution @pastaq. Sorry for giving you the run around for so long 😛

**Pull request merged**
